### DISCUSSION TITLE
2.6.7

### DISCUSSION THREAD

**Discussion by arianneorpilla** [arianneorpilla/jidoujisho#203] [Announcements]
This is a hotfix release for  [**2.6**](https://github.com/lrorpilla/jidoujisho/releases/tag/2.6.0) which includes **breaking changes for existing users**. Please refer to the [**releases page**](https://github.com/lrorpilla/jidoujisho/releases) for a full timeline of changes. 

**System Requirements**: This application has been tested to work on a variety of devices and emulators running Android 7 to 13. Please note that a significant amount of internal storage space may be taken up by the dictionary database for indexing in order to optimize query speed and performance. Try to give the application around 2-4GB of storage space. Make sure your Android WebView is updated before use. This will resolve common issues that may occur with ッツ Ebook Reader or Mokuro.

## ⚠️ Breaking Change
<details>
  <summary><b>Click To Show More</b>: For Existing Users Using The Default jidoujisho Yuuna Note Type </summary>
<br>

>There was a very poor decision with the original `jidoujisho Yuuna` template to not include the `image` and `sound` HTML tags in each card. This version will now make exporting with HTML tags on by default, with an updated default note type.
>
>If you `Clear Media`, all your media will not be detected by AnkiDroid, and you may end up deleting all your images and audio exported for the app. **Do not attempt this if you want to apply the fix to your note type. If you do so, you WILL lose media.** 
>
>You may continue use of the app if you never use `Clear Media`. However, if you wish to fix this issue, the steps are as follows. **Before doing anything, backup your** `collection.media` **folder and export your decks as** `.apkg`. Make sure you back up **both**.
>
> **Do not sync to AnkiWeb until you can confirm everything is working after the migration.**
>
>- Import your backup `.apkg` to Anki for desktop.
>- Select all `jidoujisho Yuuna` cards.
>- Find and replace `Image` field, find `(.+)` and replace with `<img src="${1}"></img>`. Use regular expressions.
>- Find and replace `Term Audio` and `Sentence Audio` field, find `(.+)` and `[sound:${1}]`. Use regular expressions.
>- Remove tags for `img` and `sound` from the back of the `jidoujisho Yuuna` note type.
>- Export from Anki and import the `.apkg` to AnkiDroid.
>- Using jidoujisho, go to the `Standard` profile and toggle on `Include image/audio HTML tags on export`.  
>
> If you have been using the `jidoujisho Yuuna` template, your cards will need the following changes above. However, so long as you do not use `Clear Media`, you will not require this migration. **Exporting with HTML tags will be on by default with every future iteration of the app moving forward.** Toggle this off with each fresh install of the app if you don't desire to migrate. If you don't have a `jidoujisho Yuuna` note type, you will not have HTML tags included with your template, and you will not require the changes above.  
>
> Again, so long as you do not use `Clear Media`, you may simply just update and continue to use the app. It is only when you will require its use that you will need to perform the fix above, as well as toggling off the option to export with HTML tags if fresh installing the app.
>
>**Due to this drastic change, this notice will be included moving forward with any future [2.6](https://github.com/lrorpilla/jidoujisho/releases/tag/2.6.0) releases**. I apologise for what this change may cause, as I understand the hard work you have put to make and collect your cards, and I want to be transparent of this issue before this causes anyone any trouble.

</details>

# 2.6.7
- Increased the dismissal speed of the dictionary pop-up upon swipe.
- Fixed an issue where audio would lag behind and be cut off when resuming the video player.
- Fixed an issue where dismissing the dictionary pop-up partially or completely would show the search result if the search completes.
- Fixed an issue where words might remain highlighted when dismissing the dictionary pop-up in ッツ Ebook Reader too fast after the search. 
- Fixed an issue where the highlighted subtitle could possibly be lost when changing the visibility of subtitle timings.

# 📲 Which file should I download?
As a rule of thumb, `arm64` should work for most modern mobile devices. `armeabi` should work on older devices. Generally, desktops and emulators should use `x86_64`.

Depending on which device you are using, download the appropriate `APK` by expanding the `Assets` drop-down below and install on your mobile phone by launching the `APK` with your file explorer.

<hr /><em>This discussion was created from the release <a href='https://github.com/lrorpilla/jidoujisho/releases/tag/2.6.7'>2.6.7</a>.</em>

**Comment by navy2016** [comment]
闪退呀，好像是ankidroid连接问题，怎么解决啊？

**Reply by arianneorpilla** [reply]
I don't study Mandarin, so I need to machine translate your question to understand it, so sorry if I don't follow.

You need to install AnkiDroid and grant permissions at startup, and it should just work.

**Comment by navy2016** [comment]
Thank you for your reply.
Click "Card Creator",A pop-up window will appear on the screen (see attachment), and then it will flash back immediately. I have to clear all the data before I can reopen jidoujisho.



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 04:33 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
I need to machine translate your question to understand it, so sorry if I don't follow.
 
You need to install AnkiDroid and grant permissions at startup, and it should just work.

**Reply by arianneorpilla** [reply]
I don't see anything attached. What version of AnkiDroid are you using?

**Comment by navy2016** [comment]
AnkiDroid：2.16.0beta1


The contents of the pop-up window are as follows:


&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;AnkiDroid error
There was an issue communicating
with AnkiDroid.Ensure that the AnkiDroid
background service is active and all
relevant app permissions are granted
in order to continue.
LAUNCH ANKIDROID&nbsp; &nbsp; CLOSE



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 05:07 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
I don't see anything attached. What version of AnkiDroid are you using?

**Reply by arianneorpilla** [reply]
I use the AnkiDroid version (2.15.6) currently available on the Play Store, seems you are using the latest pre-release.

Can you try launching AnkiDroid and then try exporting a card?

**Comment by navy2016** [comment]
Thank you for your reply.I can export the deck, but I can't find where to export a single card.



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 05:19 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
I use the AnkiDroid version (2.15.6) currently available on the Play Store, seems you are using the latest pre-release.
 
Can you try launching AnkiDroid and then try exporting a card?

**Reply by arianneorpilla** [reply]
I meant creating a card in the app. Once you launch AnkiDroid and go back to try opening the Card Creator, does it still show the pop-up?

If you force close the app, does it ask for permissions you need to grant? Make sure that jidoujisho is granted the permissions for AnkiDroid as well.

**Comment by navy2016** [comment]
Can you tell me the version of Ankⅰconnect Android that is used together? I use the latest version of Ankconnect Android, but Ankⅰconnect Android can't start, and it will automatically shut down. What should I do?thank you



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 05:19 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
I use the AnkiDroid version (2.15.6) currently available on the Play Store, seems you are using the latest pre-release.
 
Can you try launching AnkiDroid and then try exporting a card?

**Reply by arianneorpilla** [reply]
Hi @navy2016, jidoujisho does not depend on AnkiConnect Android -- it is not necessary. You only need AnkiDroid.

**Comment by navy2016** [comment]
" Once you launch AnkiDroid and go back to try opening the Card Creator, does it still show the pop-up? "


Yes


" If you force close the app, does it ask for permissions you need to grant? Make sure that jidoujisho is granted the permissions for AnkiDroid as well. "


I didn't force the app to shut down. it shut down automatically. I can't grant jidoujisho the permissions.



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 05:38 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
I meant creating a card in the app. Once you launch AnkiDroid and go back to try opening the Card Creator, does it still show the pop-up?
 
If you force close the app, does it ask for permissions you need to grant? Make sure that jidoujisho is granted the permissions for AnkiDroid as well.

**Reply by arianneorpilla** [reply]
Hi, I didn't mean that you should force close AnkiDroid. I meant if you startup jidoujisho, does it prompt you to grant AnkiDroid permissions?

Can you show me the permissions you have granted to jidoujisho? 

Again, just to clarify: to use this app, you need AnkiDroid and jidoujisho.

AnkiConnect Android is not necessary and jidoujisho does not interact with it.

**Reply by navy2016** [reply]
![IMG_20230512_094331](https://github.com/lrorpilla/jidoujisho/assets/18419494/5b94efbb-119c-47e5-8a54-cb4363490ecd)

![IMG_20230512_044651](https://github.com/lrorpilla/jidoujisho/assets/18419494/cc47290e-1001-499c-adbb-3edcbaaefc61)

**Comment by arianneorpilla** [comment]
![Screenshot_20230512-120548](https://github.com/lrorpilla/jidoujisho/assets/11363922/3ae199e7-afca-4d39-a1bb-ad5ac3eda179)
![Screenshot_20230512-120552](https://github.com/lrorpilla/jidoujisho/assets/11363922/7610c26f-831e-46f6-ac4a-e3f308120ccb)

@navy2016 Have these permissions been granted? What happens when you press `LAUNCH ANKIDROID`?

**Comment by navy2016** [comment]
Thank you for your reply.


I can't press LAUNCH ANKIDROID, because it flashes back.So I can't see your interface for setting permissions.



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 10:08 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 

 
 
@navy2016 Have these permissions been granted? What happens when you press LAUNCH ANKIDROID?

**Reply by arianneorpilla** [reply]
So AnkiDroid crashes at startup? 
Even if jidoujisho is not used?

**Reply by NoHeartPen** [reply]
> So I can't see your interface for setting permissions.

To see this settings interface, you need to turn off "MIUI Optimization" and then long-press the application icon on the desktop, click the "App info" button, and click the "Permissons" button.
为了看到这个设置界面，你需要先关闭"MIUI 优化"，然后在桌面上长按应用图标，点击"应用信息"按钮，再点击"权限"按钮。

P.S. You can use the search function in the setting app to quickly find this option by using the keyword "optimization"(But you need to enter "developer mode" first. You can search "[developer mode + Xiaomi](https://www.bing.com/search?q=%E5%BC%80%E5%8F%91%E8%80%85%E6%A8%A1%E5%BC%8F+%E5%B0%8F%E7%B1%B3)" on Bing for relevant methods.).
提示：你可以用设置的搜索功能，通过"优化"这个关键词来快速查找这个选项（但你要先进入"开发者模式"，你可以在Bing上搜索"[开发者模式+小米](https://www.bing.com/search?q=%E5%BC%80%E5%8F%91%E8%80%85%E6%A8%A1%E5%BC%8F+%E5%B0%8F%E7%B1%B3)"获取相关方法）。
![Screenshot_2023-05-12-15-53-29-292_com android settings](https://github.com/lrorpilla/jidoujisho/assets/79316356/5317aee8-b620-4515-9c88-8a00523a1f1c)

After you grant permission, you can turn on "MIUI Optimization" again.
当你授予完权限后，你可以重新开启"MIUI优化"。

**Comment by navy2016** [comment]
When using AnkiDroid alone, AnkiDroid is normal and will not crash.



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 10:30 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
So AnkiDroid crashes at startup?
 Even if jidoujisho is not used?

**Reply by arianneorpilla** [reply]
The `LAUNCH ANKIDROID` button is intended to open to the Play Store page of AnkiDroid if it is not installed.

If you have a phone that blocks Google APIs, i.e. a Chinese ROM, my guess is that it does not allow you to navigate to the store at all.

Even if you open AnkiDroid, and then open my app, it does not work?

@NoHeartPen Sorry to ask you for a favour but can you help me communicate with this user?

**Reply by NoHeartPen** [reply]
Copy that:)
I think this problem may be caused by the "MIUI Optimization" that hides some system permissions and it also makes some system permissions default values different with AOSP.

我认为此问题可能是由 “MIUI优化” 引起的，该优化隐藏了一些系统权限，并且还使某些系统权限默认值与AOSP不一样。

**Reply by NoHeartPen** [reply]
When I use [ankihelper](https://github.com/mmjang/ankihelper), which like jidoujishou does not depend on AnkiConnect Android, if I don't grant permissions correctly, I also encounter the same problem that ankihelper crashes but AnkiDriod is normal. I solve them through these articles. I hope this can be helpful.

当我使用 [ankihelper](https://github.com/mmjang/ankihelper) (它像 jidoujishou 一样不依赖 AnkiConnect Android)，如果我没有正确地授予权限，我也会遇到同样的问题：ankihelper 崩溃但 AnkiDriod 正常。我通过下面的这些文章解决了这个问题，希望能帮助你解决这个问题。

- [Anki 划词助手使用方法&常见问题 ](https://docs.qq.com/doc/DWlZRWndhWXlxa2tk)
- [酷安的帖子](https://www.coolapk.com/collection/2339801)
- [小米手机anki划词助手打开直接黑屏闪退? - NoHeartPen的回答 - 知乎](https://www.zhihu.com/question/456416422/answer/1947835029)

You can try the first document method - by turning on "Developer Settings" to turn off "MIUI Optimization" and grant relevant permissions. If clicking on the grant permission button after turning off "MIUI Optimization" doesn't work, you can try Kuan and Zhihu's answer - they mentioned an app specifically designed to solve MIUI system permission granting: Hidden Settings For MIUI

Download by [Google Play Store](https://play.google.com/store/apps/details?id=com.ceyhan.sets) or by [ApkPure](https://apkpure.com/cn/hidden-settings-for-miui/com.ceyhan.sets)

你可以先尝试第一个文档的方法——通过开启"开发者设置"来关闭"MIUI优化"进而授予相关权限。如果关闭"MIUI优化"后，点击授予权限的按钮也没有用，你可以试试酷安和知乎的回答——他们提到了一个专门解决 MIUI 系统权限授予的APP：Hidden Settings For MIUI 

软件下载方法：[Google Play Store](https://play.google.com/store/apps/details?id=com.ceyhan.sets) or by [ApkPure](https://apkpure.com/cn/hidden-settings-for-miui/com.ceyhan.sets)

**Reply by arianneorpilla** [reply]
@NoHeartPen Thanks for your help, I really appreciate it. I'll probably document these in the Troubleshooting page for the wiki.

**Comment by navy2016** [comment]
非常感谢！我想或许你是对的，不过我不打算关闭MIUI优化。关闭MIUI优化，可能会给系统带来一些不稳定。Jidoujisho的功能，我很喜欢，希望以后有机会体验！再次感谢！



---Original---
From: "John ***@***.***&gt;
Date: Fri, May 12, 2023 16:03 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




  
So I can't see your interface for setting permissions.
  
To see this settings interface, you need to turn off "MIUI Optimization" and then long-press the application icon on the desktop.
 为了看到这个设置界面，你需要关闭"MIUI 优化"，然后在桌面上长按应用图标。
 P.S. You can use the search function in the setting app to quickly find this option by using the keyword "optimization".
 提示：你可以用设置的搜索功能，通过"优化"这个关键词来快速查找这个选项。

**Reply by NoHeartPen** [reply]
After you grant permission, you can turn on "MIUI Optimization" again:)
授予完权限后，你可以重新开启"MIUI优化":)

**Comment by navy2016** [comment]
真的太热心了，谢谢你！



---Original---
From: "John ***@***.***&gt;
Date: Fri, May 12, 2023 16:38 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
When I use ankihelper, which like jidoujishou does not depend on AnkiConnect Android, if I don't give grant permissions correctly, I also meet the same problem that ankihelper crashes but AnkiDriod is normal. I solve them through these articles.I hope this can be helpful.
 
当我使用 ankihelper (它像jidoujishou一样不依赖于AnkiConnect Android)，如果我没有正确地授予权限，我也会遇到同样的问题：ankihelper崩溃但 AnkiDriod 正常。我通过下面的这些文章解决了这个问题，希望能帮助你解决这个问题。
  
Anki 划词助手使用方法&amp;常见问题 
 
酷安的帖子
 
小米手机anki划词助手打开直接黑屏闪退? - NoHeartPen的回答 - 知乎

**Comment by navy2016** [comment]
好的，谢谢了，有空试试😊



---Original---
From: "John ***@***.***&gt;
Date: Fri, May 12, 2023 16:58 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lrorpilla/jidoujisho] 2.6.7 (Discussion #203)




 
After you grant permission, you can turn on "MIUI Optimization"
 授予完权限后，你可以重新开启"MIUI优化":)

**Comment by Getroekah** [comment]
I have the same issue
I have Xiaomi 10c with miui 13 
It keeps saying there was an issue communicating with ankidroid despite it having all permissions and running in the background also disabled miui optimization which reset all the permissions to all my apps but that still didn't work D:
